obelisk/cpp/zoo/hardware/sensing/zed/zed2_sensors_interface.h

```diff
@@ -0,0 +1,420 @@
+#pragma once
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <fstream>
+#include <msg_conversions.h>
+#include <mutex>
+#include <obelisk_sensor_msgs/msg/obk_image.hpp>
+#include <obelisk_std_msgs/msg/u_int8_multi_array.hpp>
+#include <rclcpp/callback_group.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/timer.hpp>
+#include <rclcpp_lifecycle/lifecycle_node.hpp>
+#include <set>
+#include <sl/Camera.hpp>
+#include <stdexcept>
+#include <unsupported/Eigen/CXX11/Tensor>
+#include <yaml-cpp/yaml.h>
+
+#include "obelisk_sensor.h"
+
+/**
+ * @brief Class for managing ZED2 cameras using the Obelisk framework.
+ */
+class ObeliskZed2Sensors : public obelisk::ObeliskSensor {
+
+  public:
+    /**
+     * @brief Construct a new ObeliskZed2Sensors object.
+     *
+     * @param node_name Name of the ROS2 node.
+     */
+    ObeliskZed2Sensors(const std::string& node_name) : obelisk::ObeliskSensor(node_name) {
+        // Declare parameters
+        this->declare_parameter("params_path_pkg", "");
+        this->declare_parameter("pub_period", 0.0);
+
+        // Get parameter values
+        pub_period_                 = this->get_parameter("pub_period").as_double();
+        std::string params_path_pkg = this->get_parameter("params_path_pkg").as_string();
+        std::string _params_path    = this->get_parameter("params_path").as_string();
+
+        std::string params_path;
```

Comment:
I've been trying to use std::filesystem::path for paths. Tbh its effectively a string, but it has the semantic meaning. Also you can use functions like `std::filesystem::exists` with it to check if the path exists.

Reply:
resolved.